### PR TITLE
Fix too agressive logging-format-interpolation

### LINKED ISSUE #572: logging-format-interpolation: too aggressive
Originally reported by: **BitBucket: [sam-s](http://bitbucket.org/sam-s), GitHub: @sam-s?**

---

I get this warning:

```
(W1202, logging-format-interpolation, my_func) Use % formatting in logging functions but pass the % parameters as arguments
```

when `my_func` does something like

```
logger.info("Read {l:,d} rows".format(l=n))
```

note that this is _not_ equivalent to 

```
logger.info("Read %d rows",n)
```

because `%d` does not insert commas as in `123,456,789`.
I suggest that this warning is only retained for `%` conversions, e.g.:

```
logger.info("Read %d rows" % (n))
```

but _not_ for `"...".format()`.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/572

### PR DESCRIPTION
**Pull request by rogalski**
Fixes #572

### CODE REVIEW

File: pylint/checkers/logging.py [hunk 1]
```diff
@@ -232,6 +232,23 @@ def _check_format_string(self, node, format_arg):
...
+    Args:
+        node (astroid.node_classes.NodeNG): AST node to check
+    Returns:
+        bool: True if inferred string uses complex formatting, False otherwise
+    """
+    inferred = utils.safe_infer(node)
+    if inferred is None or not isinstance(inferred.value, six.string_types):
+        return True
+    for _, _, format_spec, _ in string.Formatter().parse(inferred.value):
+        if format_spec:
```
Comment (rogalski) [inline pylint/checkers/logging.py]: maybe some simple formatter specs may be detected and converted to suggestions (e.g. `!r` -> `%r`)

File: pylint/checkers/logging.py [hunk 2]
```diff
@@ -56,32 +57,28 @@
     }
 
 
-CHECKED_CONVENIENCE_FUNCTIONS = set([
-    'critical', 'debug', 'error', 'exception', 'fatal', 'info', 'warn',
-    'warning'])
+CHECKED_CONVENIENCE_FUNCTIONS = {'critical', 'debug', 'error', 'exception',
```
Comment (PCManticore) [inline pylint/checkers/logging.py]: Let's use 4 spaces hanging indentation here, as it was previously.

File: pylint/checkers/logging.py [hunk 3]
```diff
@@ -232,6 +232,23 @@ def _check_format_string(self, node, format_arg):
             self.add_message('logging-too-few-args', node=node)
 
 
+def is_complex_format_str(node):
+    """ Checks if node represents a string with complex formatting specs.
```
Comment (PCManticore) [inline pylint/checkers/logging.py]: There's an extra space at the start of this docstring

File: pylint/checkers/logging.py (hunk 1 above)
Comment (PCManticore) [inline pylint/checkers/logging.py]: That would be interesting. Let's do it in a separate PR though.
